### ISSUE TITLE
Correct resizing of images

### ISSUE THREAD

**Issue by thanksforthefish** [facebookresearch/detectron2#4662]
I train on large images (approx. 4000 x 4000 px) and downsize them in a custom mapper. However, this seems to have no effect on the training time. The training time is the same if I do not use the custom mapper.

## Instructions To Reproduce the Issue:
I use the following custom mapper:
```
def custom_mapper(dataset_dict):
    dataset_dict = copy.deepcopy(dataset_dict)  # it will be modified by code below
    image = utils.read_image(dataset_dict["file_name"], format="BGR")
    transform_list = [
        T.ResizeShortestEdge(800)
    ]
    image, transforms = T.apply_transform_gens(transform_list, image)
    dataset_dict["image"] = torch.as_tensor(image.transpose(2, 0, 1).astype("float32"))

    annos = [
        utils.transform_instance_annotations(obj, transforms, image.shape[:2])
        for obj in dataset_dict.pop("annotations")
        if obj.get("iscrowd", 0) == 0
    ]
    instances = utils.annotations_to_instances(annos, image.shape[:2])
    dataset_dict["instances"] = utils.filter_empty_instances(instances)
    return dataset_dict
```
This is my Trainer train loader:
```
    @classmethod 
    def build_train_loader(cls, cfg): 
        mapper = None
        # mapper = custom_mapper
        data_loader = build_detection_train_loader(cfg, mapper=mapper)   
        return  data_loader
``

## Expected behavior:
I expected the training time to be reduced. If I replace the images with downscaled images, the training time is significantly reduced: I use a resize factor of .25 and the time is approx. reduced by the same factor. - as expected.

Could someone please tell me how to correctly downsize the images?

**Comment by ryouchinsa** [comment]
Although this is not an open source program, RectLabel can import the COCO JSON file, resize images and annotations, and export to the COCO JSON file again.
https://rectlabel.com/help#resize_images

**Comment by jvdgoltz** [comment]
My guess is that you also need to increase the batch size, otherwise there will not be much gain in reducing the image resolution. If you train on half the image resolution, you can approximately quadruple the batch size (cause GPU memory scales roughly quadratically with the resolution), that should give you a good increase in speed.

Keep in mind that you also have to change the learning rate if you change the batch size
